patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -5682,19 +5682,52 @@
  				return false;
  
  			for (int i = x - 1; i <= x + 1; i++) {
-@@ -38067,6 +_,7 @@
+@@ -38053,6 +_,28 @@
+ 			}
+ 			catch (Exception exception) {
+ 				FancyErrorPrinter.ShowFileSavingFailError(exception, playerFile.Path);
++				// If there is a problem in saving, (kick out the player and) show message
++				var errorBuilder = new StringBuilder();
++				errorBuilder.AppendLine(Language.GetTextValue("An error occurred when saving the player.")); // TODO use key
++				errorBuilder.AppendLine(Language.GetTextValue("The original Error below:")); // TODO use key
++				errorBuilder.Append(exception);
++
++				var flag = false;
++				if (!flag && !Main.gameMenu) {
++					if (Main.netMode == NetmodeID.SinglePlayer) {
++						WorldFile.SaveWorld();
++					}
++
++					// Single & Multiplayer will quit. Server continues to run.
++					if (Main.netMode != NetmodeID.Server) {
++						WorldGen.JustQuit();
++						if (Main.netMode == NetmodeID.SinglePlayer) {
++							Main.menuMode = MenuID.Title;
++						}
++					}
++				}
++
++				Utils.ShowFancyErrorMessage(errorBuilder.ToString(), Main.menuMode); // Return to the previous menu.
```

Comment:
Suggest:
`Utils.ShowFancyErrorMessage($"{Language.GetTextValue("tModLoader.PlayerSaveFail")}\n\n{e}", Main.menuMode)`

`"PlayerLoadWorldFail": "An error occurred while saving the player"`